manager/integration/tests/test_ha.py

```diff
@@ -2619,108 +2619,143 @@ def test_replica_failure_during_attaching(settings_reset, client, core_api, volu
     common.wait_for_disk_update(client, node.name, 1)
 
 
-@pytest.mark.skip(reason="TODO") # NOQA
-def test_engine_image_missing_on_some_nodes():
+def prepare_engine_not_fully_deployed_evnironment():
+    """
+    1. Taint node-1 with the taint: key=value:NoSchedule
+    2. Delete the pod on node-1 of the engine image DaemonSet.
+       Or delete the engine image DaemonSet and wait for Longhorn
+       to automatically recreates it.
+    3. Wait for the engine image CR state become deploying
     """
-    This e2e test follows the manual test steps at:
-    https://github.com/longhorn/longhorn/issues/2081#issuecomment-783747541
 
-    Steps:
 
-    Preparation:
-    1. Set up a backup store
-    2. Let's name the 3 nodes: node-1, node-2, node-3
+def prepare_engine_not_fully_deployed_evnironment_with_volumes():
+    """
+    1. Create 2 volumes, vol-1 and vol-2 with 3 replicas
+    2. Taint node-1 with the taint: key=value:NoSchedule
+    3. Attach vol-1 to node-1. Change the number of replicas of vol-1
+       to 2. Delete the replica on node-1
+    4. Delete the pod on node-1 of the engine image DaemonSet.
+       Or delete the engine image DaemonSet and wait for Longhorn
+       to automatically recreates it.
+    5. Wait for the engine image CR state become deploying
+    """
+
 
-    Case 1: Test volume operations when engine image DaemonSet is miss
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_engine_image_miss_scheduled_perform_volume_operations():
+    """
+    Test volume operations when engine image DaemonSet is miss
     scheduled
+
     1. Create a volume, vol-1, of 3 replicas
-    2. Create another volume, vol-2, of 3 replicas
-    3. Taint node-1 with the taint: key=value:NoSchedule
-    4. Verify that we can attach, take snapshot, take a backup,
+    2. Taint node-1 with the taint: key=value:NoSchedule
+    3. Verify that we can attach, take snapshot, take a backup,
        expand, then detach vol-1
+    """
+    pass
 
-    Case 2: Test volume operations when engine image DaemonSet is not fully
+
+@pytest.mark.skip(reason="TODO") # NOQA
+def test_engine_image_not_fully_deployed_perform_volume_operations():
+    """
+    Test volume operations when engine image DaemonSet is not fully
     deployed
-    1. Continue from case #1
-    2. Attach vol-1 to node-1. Change the number of replicas of vol-1
-       to 2. Delete the replica on node-1
-    3. Delete the pod on node-1 of the engine image DaemonSet.
-       Or delete the engine image DaemonSet and wait for Longhorn
-       to automatically recreates it.
-    4. Wait for the engine image CR state become deploying
-    5. Verify that functions (snapshot, backup, detach) are working ok
+
+    1. Perform prepare_engine_not_fully_deployed_evnironment_with_volumes
```

Comment:
Prerequisite:
Prepare system for the test by calling the method `prepare_engine_not_fully_deployed_evnironment_with_volumes` to have 2 volumes, tainted node and not fully deployed engine.

Reply:
Updated!